packages/experimental/Shadow/src/Shadow.tsx

```diff
@@ -22,12 +21,95 @@ export const Shadow = stagedComponent((props: ShadowProps) => {
       }
     }
 
-    const childWithKeyShadow = React.cloneElement(child, mergeProps(child.props, { style: shadowTokenStyleSet.key }));
+    const { style: childStyle = {}, ...restOfChildProps } = child.props;
+
+    const shadowViewStyleProps = getStylePropsForShadowViews(childStyle, shadowTokenStyleSet);
+    const innerShadowViewProps = mergeProps(restOfChildProps, shadowViewStyleProps.inner);
+    const outerShadowViewProps = mergeProps(final, shadowViewStyleProps.outer);
+
+    const childWithInnerShadow = React.cloneElement(child, innerShadowViewProps);
 
-    return <View {...mergedProps}>{childWithKeyShadow}</View>;
+    return <View {...outerShadowViewProps}>{childWithInnerShadow}</View>;
   };
 });
 
+const getStylePropsForShadowViews = memoize(getStylePropsForShadowViewsWorker);
+function getStylePropsForShadowViewsWorker(childStyleProps: any, shadowTokenStyleSet: any) {
```

Comment:
`shadowTokenStyleSet` should at least have a well-defined type.

It would be nice if we could not have `any` for childStyleProps, too, but that's probably much harder to accomplish. You could try restricting to `StyleProp<ViewStyle>` but TS might complain

Reply:
There's also the type `ShadowStyleIOS` in the RN-types repo:

```
export interface ShadowStyleIOS {
    shadowColor?: ColorValue | undefined;
    shadowOffset?: { width: number; height: number } | undefined;
    shadowOpacity?: number | undefined;
    shadowRadius?: number | undefined;
}
```

Reply:
I'm pretty sure we have a type defined for shadow styles in our theming package? I just don't remember what it is

Reply:
Not for the actual RN type, we have the "design" type. In the future we can perhaps just have the pipeline convert the "design type" to the RN props so we don't have to know to do stuff like divide blur by 2 on iOS, but 🤷🏾

```
/**
 * Base structure of a shadow style, which has x and y offset, blur amount, and color specified
 */
export interface ShadowValue {
  x: number;
  y: number;
  blur: number;
  color: ColorValue;
}

```

Reply:
But yeah, let's not use `any`

Reply:
There wasn't really one that fit what I needed exactly but just moved the getShadowStyle call into the getStyleProps method so that I can just pass in the shadowToken

Reply:
ViewStyle seems to work for childStyleProps